src/yaml.cr

```diff
@@ -147,4 +148,28 @@ module YAML
   def self.dump(object, io : IO)
     object.to_yaml(io)
   end
+
+  # Determines if the scaler is reserved
+  def self.reserved_scalar(value : String)
```

Comment:
I'm not sure I like this method much. For example it tries to linearize the scalar value (using a non-standard naming: `infinity`, `NaN`), but sometimes returns a detected tag (`int`, `float`) which means the integer/float number will have to be parsed *again*.